### ISSUE TITLE
Trust Wallet 6.2 iOS

### ISSUE THREAD

**Issue by mhorsley30896** [aktionariat/walletconnect-bridge#1]
Trust wallet updated their app in iOS to 6.2. This implementation of the bridge no longer works for connecting to Trust Wallet via wallet connect. I can confirm that reverting back to the official https://bridge.wallectconnect.org doesn't have this issue, so I do believe it is an issue with this implementation (though I agree with you the official server is very intermittent!)

**Comment by mhorsley30896** [comment]
This is the output from my instance when attempting to connect to trust wallet 6.2. In the app the connection is successful, however when you return to the browser (as directed to do so) the connection is closed

```
root@grv:~/walletconnect-bridge# java -cp target/bridge.jar:target/dependency/* com.aktionariat.bridge.BridgeServer
[main] INFO com.aktionariat.bridge.BridgeServer - WalletConnect Bridge Java Edition 0.1 started on port: 8887
[WebSocketWorker-11] INFO com.aktionariat.bridge.BridgeServer - /127.0.0.1 opened a connection to us
[WebSocketWorker-11] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"863eff01-12f2-46ed-ad74-ca060498ef19","type":"pub","payload":"{\"data\":\"26202d28574728260834a442d674a80b568ed369dfa1e2fcd52e8176e10693a53614b112404a50eecf1e6e33acc0fe1b969cf03910f92318c78dfb31c61bdf8d92c9a2c31a63979d854c640f58a742923561be33df7508f874ea9f12a0e7ae7a58af5c940e933be0a8a71d6e3ab58494d93544b089e69e4aa66210e96dd79ab0ead8aee4b93406e2347fa21a8d91397128ba4001e62bb1e2186ca1edcea3013f8b8bcef7c929aa14cb3e2ec5a360b6dacccad378c4cd8e79d29f387d4a19ea8d8aa928bf31b6a80667b89d763e056686bd6420dce1e37eda8e0e699c084359d9082c952052084381cf4fcca250e0df0cc135f5f8f470baa2e9a6fa56004a92a4daaac08d3b1a2682ed2f6565942a6d89d6f4ae6ab4249b12f82d78ae2b0f8bb671282c0fa0549b5174ad28ba78c9ff16c6f7b55ecdee171a33cea9b0c3f73c9626104a8342c4944ff6dcc8ba6536a5369565aed50e78401c95a5b780a9b95c0811a4e7bbe9dc41b8796cec0cef89942930cecbba03ea5761d20b4d5822918a5a1d78812222a9bb54be46d54f8863e0b832247fffc599a888e65e98b304dc8e188c58c0cd4e34ef61c0baa107d761f12a2528d4f0ee16a77a3e6b4e8d4074ce9f8cfff8464a891ebea5ef14281aa7cdbf73e24a708122272c978117c10f155188c870beb3960feaf5b5c1e6475423a7bdcf6b9f4049a804f7f46a9a017d9b5a691e8eae5649a930f662e2a8672b941a02\",\"hmac\":\"537affe7bd6874f558bc211ad2cae153cc25da232567f8a727ba7dc0218ebb31\",\"iv\":\"fd17cd948d625a38dca0ba290d1887cc\"}","silent":true} from /127.0.0.1
[WebSocketWorker-11] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"b1e6c6ff-9ab1-4c56-99ae-2bcdf0481753","type":"sub","payload":"","silent":true} from /127.0.0.1
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - /127.0.0.1 opened a connection to us
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"863eff01-12f2-46ed-ad74-ca060498ef19","type":"sub","payload":""} from /127.0.0.1
[WebSocketWorker-13] INFO com.aktionariat.bridge.BridgeServer - /127.0.0.1 opened a connection to us
[WebSocketWorker-14] INFO com.aktionariat.bridge.BridgeServer - /127.0.0.1 opened a connection to us
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"b1e6c6ff-9ab1-4c56-99ae-2bcdf0481753","type":"sub","payload":""} from /127.0.0.1
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Implicitely acking Bridge to org.java_websocket.WebSocketImpl@495bffb6 since Fri Jan 24 01:01:34 UTC 5997 from /127.0.0.1:41964
Replacing subscriber org.java_websocket.WebSocketImpl@51d69f6b with  /127.0.0.1:41964
[WebSocketSelector-15] INFO com.aktionariat.bridge.BridgeServer - null closed the connection, status 1000
[WebSocketWorker-13] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"3cfb575d-9779-4ae0-96bc-1f84cfe99b5a","type":"sub","payload":""} from /127.0.0.1
[WebSocketWorker-14] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"eb41a431-b01e-4130-bda1-c532e03fc760","type":"sub","payload":""} from /127.0.0.1
[WebSocketWorker-14] INFO com.aktionariat.bridge.BridgeServer - Received {"payload":"{\"iv\":\"a1d6f272bd24c491f93aaa93ea22a33f\",\"data\":\"220ba1bff69fb85f2785c7b7b253419d85da01b4fc8fc4f10f854b09b38331faa0959c7dc01a27ff32906a9be7162fcaaebdff4b6325d30e9b6934bc6b432a02eb2629b7ac244e056d7623dd2151cea74c6db169e131741873a3d470b07fcfa6eff5317de3b6b4f2040870bc9c24806cab1b79149cd3cca8057040e4f76ea7d2\",\"hmac\":\"c5a610528c732776cbe8d4fcd01ef91f1adf78ca687237eaa581c1489402b93c\"}","topic":"eb41a431-b01e-4130-bda1-c532e03fc760","type":"pub"} from /127.0.0.1
[WebSocketWorker-14] INFO com.aktionariat.bridge.BridgeServer - Implicitely acking Bridge to org.java_websocket.WebSocketImpl@799258d9 since Mon Jan 27 00:57:51 UTC 5997 from /127.0.0.1:41968
[WebSocketSelector-15] INFO com.aktionariat.bridge.BridgeServer - null closed the connection, status 1006
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Received {"payload":"{\"iv\":\"65d7e5f884c1ae7c601d9b1e84a23673\",\"data\":\"c19f90f3d060ad90280b451fefd783a8bfda853f812c369e396221d0947cbeb36ee2b7c098d3d9a92843133218712720707d9020b0387c72f5cae8839cab4bbc21f6c8d27b885addc9966ac713b315e9ce8858905d38f054effc4ff19a7ab405fc6e6c3ccbbf090dab16402e9898eb0cfc5559349a43cb44657654d5d30c89d8d4553332c68ff78935a976dafe252f18922037df4270a915f3aa37b6a92ef7af99c1e3b6917b81d358b949585c72206116af9dffc32fb5b81454b4fd0a9bc527c222135b808d2a8ec6958a20e880f649c7be821e6f5806ce17bc88768ae40bfc78135360560171b4c140d5a4b974404077e56bd77d662ba526fc99c71d0855f197b01d49d39f2c87b63726001bb69c34399ec62e4c3120b6c33fa1d90c4980c6612a775f41f09cb9e2da68a72a4279f9bfd7834e651c190260d9ef7dc9fda893cca2380b6f7b134f894edba07ab7aedb5fc892a6bf98f6a9688fdc18804355994dff6dc3a03b9949b404ea6ec07242266a4778d109faea333f45670255f1d5f3089b71a72504e955c653dfc38cbf6d6d06c7b55b2949e0a1158b415715080ccf35e9b351bde522c93eb2197b2ea4ded7cbb1b2d0c8bb5d62e3067b3033a65f2119024dd52ba98fb6e739fa59f83aba19a9cd7c561a82cde62b4b5f29e06efe826eec72069f5663f653bfe00985fa6e9f190b0f14a88dc0876127a890c2c365991fd2fc991c13150169069002e4e159c0a0509cd76db4fcb9d603118bb2ce27c72d721884738f2f84cbec2c73abdaa83c965f5e09ec57ba4252abdfaf0112edb9275aed8671bb1e3611817e05dd543054\",\"hmac\":\"75d7ea521765fdc424dcbe0ff30bbf7e8f10279898e934b1987297f3110c7f7c\"}","topic":"b1e6c6ff-9ab1-4c56-99ae-2bcdf0481753","type":"pub"} from /127.0.0.1
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Implicitely acking Bridge to org.java_websocket.WebSocketImpl@495bffb6 since Fri Jan 24 02:15:54 UTC 5997 from null
[WebSocketWorker-12] INFO com.aktionariat.bridge.BridgeServer - Implicitely acking Bridge to org.java_websocket.WebSocketImpl@495bffb6 since Sun Jan 26 07:07:17 UTC 5997 from null
[WebSocketSelector-15] INFO com.aktionariat.bridge.BridgeServer - null closed the connection, status 1006
[WebSocketWorker-11] INFO com.aktionariat.bridge.BridgeServer - /127.0.0.1 opened a connection to us
[WebSocketWorker-11] INFO com.aktionariat.bridge.BridgeServer - Received {"topic":"b1e6c6ff-9ab1-4c56-99ae-2bcdf0481753","type":"sub","payload":""} from /127.0.0.1
[WebSocketSelector-15] INFO com.aktionariat.bridge.BridgeServer - null closed the connection, status 1006
[WebSocketSelector-15] INFO com.aktionariat.bridge.BridgeServer - null closed the connection, status 1006
```

**Comment by luziusmeisser** [comment]
Reorganized the behaviour of our server and released version 0.2.

See commit comment for details.
